Sources/Datadog/URLSessionAutoInstrumentation/DDURLSessionDelegate.swift

```diff
@@ -53,8 +55,7 @@ open class DDURLSessionDelegate: NSObject, URLSessionTaskDelegate, URLSessionDat
     ///
     /// - Parameter additionalFirstPartyHosts: these hosts are tracked **in addition to** what was
     ///             passed to `DatadogConfiguration.Builder` via `trackURLSession(firstPartyHosts:)`
-    @objc
-    public convenience init(additionalFirstPartyHosts: Set<String>) {
+    public convenience init(additionalFirstPartyHosts: FirstPartyHosts) {
         self.init(in: defaultDatadogCore, additionalFirstPartyHosts: additionalFirstPartyHosts)
     }
```

Comment:
This will be a breaking change in public API, and we can't do it:
```diff
- Set<String>
+ [String: Set<TracingHeaderType>]
```
To avoid it, we can follow the same pattern as we did in `DatadogConfiguration`: introduce a new API next to the old one and call it internally to have single control flow.